lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,7 @@ def word_symbol_pair?(pair)
           if key.sym_type?
             sym_name = key.loc.expression.source
 
-            sym_name !~ /\A:["']/
+            sym_name !~ /\A:["']/ && !sym_name.end_with?('=')
```

Comment:
@bbatsov I agree, but I need some help getting there.

For example, the spec in spec/rubocop/cop/style/hash_syntax_spec.rb:281 (“accepts hash rockets when keys have whitespaces in them”) gives me this to work with:
```rb
p pair
# (pair
#   (sym :"t o")
#   (int 0))
p key.loc.expression.source
# ":\"t o\""
p key.loc.expression.source.to_sym
# :":\"t o\""
p key.to_a.first
# :"t o"
p key.to_a.first.to_s
# "t o"
```
For line 50 (“accepts hash rockets when symbol keys end with =”), I get:

```rb
p pair
# (pair
#   (sym :a=)
#   (int 0))
p key.loc.expression.source
# ":a="
p key.loc.expression.source.to_sym
# ::a=
p key.to_a.first
# :a=
p key.to_a.first.to_s
# "a="
```
So I don’t really know which values to compare.